### ISSUE TITLE
rpi4: USB Keyboard not working, udhcpc can't get a lease in initrd

### ISSUE THREAD

**Issue by pschyska** [Tow-Boot/Tow-Boot#181]
During the installation, my USB Keyboard started to work only after the kernel had booted the installer. I couldn't enter e.g. tow-boot boot menu, or select a boot option from the install iso.

I rebooted multiple times and after getty dropped me into a shell, the keyboard either started working right away, or needed a re-plug. Afterwards it continued working without issue, and I finished the OS installation.

Now, I can't get the keyboard to work at all to enter my luks passphrase. I tried two different keyboards, with and without different USB hubs, powered and unpowered ones.

Additionally, udhcpc doesn't get a lease so I can't unlock via SSH (as intended). With https://github.com/pftf/RPi4 both worked without any issues.

Any idea what's happening, and what I can do to fix it? As I can't enter the tow-boot firmware menu I can't make changes there, but is there anything I could try by modifying the build and re-building myself?

**Comment by pschyska** [comment]
I updated EEPROM via Raspberry OS to https://github.com/raspberrypi/rpi-eeprom/commit/e4f90a89e62051ff658c40c57150d13c7e67553b beforehand.

**Comment by pschyska** [comment]
I switched to https://github.com/pftf/RPi4 and had the same keyboard issues, until I realized I have to make available "pcie-brcmstb" and "reset-raspberrypi" modules. Because the keyboard did work in the nixos installer image, I'm 90% sure that this was my issue, so I'm gonna close this - the keyboard not working in the bootloader is still a problem but in another issue https://github.com/Tow-Boot/Tow-Boot/pull/178 was mentioned for usb keyboard support so this might solve that?
